upgrades/upgrade_testing_framework/cluster_management/cluster_objects.py

```diff
@@ -0,0 +1,6 @@
+from dataclasses import dataclass
+
+@dataclass
+class ClusterSnapshot:
```

Comment:
Not sure I agree on this one.  I mean, it's all from the perspective of the thing that uses the object.  In this case, what matters is to keep track of the snapshot id and the repo it's in so that it can be restored later.

Also, snapshots live in repos, right?  And it's possible that you could have two different snapshots with the same ID in two different repos?  But you couldn't have two snapshots with the same ID in the same repo?  So the combination of the repo and the snapshot ID is the primary key for the snapshot.

You could make an object for the snapshot, and an object for the repo, but it feels like the snapshot object would contain a reference to the repo object, which leads us back to where we started.

Reply:
Okay I think I misinterpreted what you intended here. You are viewing this as an object to represent a single snapshot not an object to represent all things snapshot for a cluster. I think I'm okay with this then